src/app/views/charts/histogram.js

```diff
@@ -1,165 +1,179 @@
+    // },
+
+    // highlightBar: function (bars, context) {
+    //   const ctxValue = context.selected_value;
+
+    //   const quantiles = context.colorScale.quantiles
+    //     ? [...context.colorScale.quantiles()]
+    //     : [...context.colorScale.domain()];
+
+    //   const highlightIndex =
+    //     ctxValue !== null
+    //       ? context.findQuantileIndexForValue(ctxValue, quantiles)
+    //       : null;
+
+    //   bars.classed('highlight', function (d, i) {
+    //     return i === highlightIndex;
+    //   });
+    // },
+
+    // render: function () {
+    //   const colorScale = this.colorScale;
+    //   const isThreshold = colorScale.quantiles ? false : true;
+
+    //   const gradients = this.gradients;
+    //   const counts = _.pluck(gradients, 'count');
+    //   const height = this.height;
+
+    //   const yScale = d3
+    //     .scaleLinear()
+    //     .domain([0, d3.max(counts)])
+    //     .range([0, this.height]);
+
+    //   const xScale = d3
+    //     .scaleOrdinal()
+    //     .domain(d3.range(0, this.slices))
+    //     .rangeBands([0, this.width], 0.2, 0);
+
+    //   // threshold types use rounded bands for convienence
+    //   if (isThreshold) {
+    //     xScale.rangeRoundBands([0, this.width], 0.1, 0);
+    //   }
+
+    //   const bardata = xScale.domain().map((d, i) => {
+    //     return {
+    //       ...gradients[i],
+    //       idx: i,
+    //       data: d,
+    //       xpos: xScale(d) + xScale.rangeBand() / 2
+    //     };
+    //   });
+
+    //   const filterValueForXpos = d3.scale
+    //     .linear()
+    //     .range(this.filterRange)
+    //     .domain([0, this.width]);
+
+    //   // make scale available to caller
+    //   this.xScale = xScale;
+
+    //   // draw
+    //   const bars = this.g.selectAll('rect').data(bardata, function (d) {
+    //     return d.color;
+    //   });
+
+    //   bars.enter().append('rect');
+
+    //   bars
+    //     .style('fill', (d, i) => {
+    //       // not on a continous scale
+    //       // so just need the color from data
+    //       if (isThreshold) return d.color;
+
+    //       // mapping the color continously
+    //       // so need to calculate the color for
+    //       // this xpos
+    //       //
+    //       return colorScale(filterValueForXpos(d.xpos));
+    //     })
+    //     .attr({
+    //       width: () => {
+    //         return xScale.rangeBand();
+    //       },
+    //       'stroke-width': 0,
+    //       height: d => yScale(d.count),
+    //       x: (d, i) => xScale(d.data),
+    //       y: d => height - yScale(d.count)
+    //     });
+
+    //   bars.exit().remove();
+
+    //   bars.call(this.highlightBar, this);
+
+    //   this.g
+    //     .selectAll('rect')
+    //     .filter((bucket, index) => {
+    //       return bucket.current === index;
+    //     })
+    //     .classed('current', true);
+
+    //   return this.el;
+    // }
```

Comment:
Temporarily commented out for errors. Seems to only be used in the building scorecard. Will reassess later.